### PR TITLE
feat: `kiln validate` checks for valid yaml and release version constaint syntax

### PR DESCRIPTION
**Pull request by crhntr**
This implementation checks for valid Kilnfile and Kilnfile.lock YAML syntax. It also ensures the release constraint syntax is valid.

### CODE REVIEW

File: commands/validate.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/pivotal-cf/jhanda"
+	"github.com/pivotal-cf/kiln/internal/cargo"
+	"gopkg.in/src-d/go-billy.v4"
+	"gopkg.in/src-d/go-billy.v4/osfs"
+	"strings"
+)
+
+type Validate struct {
+	Options struct {
+		Kilnfile       string   `short:"kf" long:"kilnfile" default:"Kilnfile" description:"path to Kilnfile"`
```
Comment (ciriarte) [inline commands/validate.go:15]: **suggestion:** I like examples in my help descriptions.

File: commands/validate.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		VariablesFiles []string `short:"vf" long:"variables-file" description:"path to variables file"`
```
Comment (ciriarte) [inline commands/validate.go:16]: **question:** do we have examples of var files?

**Review by ciriarte — APPROVED** [review]
Beautiful work.

**suggestion:** let's update the README with the new command.

File: commands/validate.go (hunk 1 above)
Comment (crhntr) [reply commands/validate.go:15]: Maybe we can create an issue for this so we do it across all the flags.

File: commands/validate.go (hunk 2 above)
Comment (crhntr) [reply commands/validate.go:16]: Yes. Kiln fetch uses them for s3 credentials. The README has an example.

**Pull request merged**